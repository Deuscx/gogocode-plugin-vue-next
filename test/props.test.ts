import { resolve } from 'path'
import { describe, expect, it } from 'vitest'
import $ from 'gogocode'
import propsTransform from '../src/rules/props'

describe('should', () => {
  it('generate props', () => {
    const ast = $.loadFile(resolve(__dirname, './examples/props.vue'), {
      parseOptions: { language: 'vue' },
    })
    propsTransform(ast)
    expect(ast.generate()).toMatchSnapshot()
  })
})
