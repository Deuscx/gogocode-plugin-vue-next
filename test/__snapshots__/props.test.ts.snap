// Vitest Snapshot v1

exports[`should > generate props 1`] = `
"
<script lang=\\"ts\\">
import { defineComponent, defineProps } from 'vue'
export default defineComponent({
  name: 'Test',

  setup() {
    return {
      props: '1',
    };
  }
})
const props = defineProps({
  width: {
    type: Number,
    default: 446,
  },
  title: {
    type: String,
    default: '',
  },
})
</script>
"
`;
